src/components/shared/passphraseRenderer/index.js

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import fillWordsList from 'bitcore-mnemonic/lib/words/english';
+import grid from 'flexboxgrid/dist/flexboxgrid.css';
+import { withTranslation } from 'react-i18next';
+import styles from './passphraseRenderer.css';
+import { PrimaryButton, TertiaryButton } from '../../toolbox/buttons/button';
+
+class PassphraseRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      indexes: [2, 9],
+      fieldSelected: undefined,
+      chosenWords: {},
+      options: {},
+      isCorrect: false,
+      hasErrors: false,
+      values: [],
+    };
+
+    this.handleConfirm = this.handleConfirm.bind(this);
+  }
+
+  componentDidMount() {
+    const { indexes } = this.state;
+    const values = this.props.passphrase.split(' ');
+
+    const options = this.assembleWordOptions(values, indexes);
+    this.setState({
+      ...this.state,
+      options: {
+        [indexes[0]]: options[0],
+        [indexes[1]]: options[1],
+      },
+      values,
+      fieldSelected: indexes[0],
+    });
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
+  }
+
+  handleConfirm() {
+    const { chosenWords, indexes } = this.state;
+    const { values } = this.state;
+
+    const answers = Object.values(chosenWords);
+    const isCorrect = answers.filter((answer, index) => answer === values[indexes[index]])
+      .length === 2;
+
+    const cb = isCorrect
+      ? () => this.props.nextStep()
+      : () => this.setRandomIndexesFromPassphrase(2);
+
+    this.setState({
+      isCorrect,
+      hasErrors: !isCorrect,
+    });
+
+    this.timeout = setTimeout(cb, 1500);
+  }
+
+  setRandomIndexesFromPassphrase(qty) {
+    const { values } = this.state;
+    let idxs = values.map((w, index) => index);
+    const indexes = [...Array(qty)]
+      .map(() => {
+        const index = idxs[Math.floor(Math.random() * idxs.length)];
+        idxs = [...idxs.slice(0, index), ...idxs.slice(index + 1)];
+        return index;
+      })
+      .sort((a, b) => a - b);
+    const options = this.assembleWordOptions(values, indexes);
+
+    this.setState({
+      options: {
+        [indexes[0]]: options[0],
+        [indexes[1]]: options[1],
+      },
```

Comment:
These two look very similar (not DRY):
https://github.com/LiskHQ/lisk-hub/blob/a2f966a9867a3dd7985f41d47bbe8ffbb63a04a8/src/components/shared/passphraseRenderer/index.js#L74-L80
https://github.com/LiskHQ/lisk-hub/blob/a2f966a9867a3dd7985f41d47bbe8ffbb63a04a8/src/components/shared/passphraseRenderer/index.js#L28-L34

It could be resolved if `this.assembleWordOptions` already returned `options` in the format that is used in `state`